editor/src/components/inspector/sections/event-handlers-section/event-handlers-section.tsx

```diff
@@ -42,15 +43,58 @@ export const EventHandlerControl = betterReactMemo(
   },
 )
 
-export const EventHandlersSection = betterReactMemo('EventHandlersSection', () => {
-  const { value, onUnsetValues } = useInspectorInfo<DOMEventHandler, ParsedValues<DOMEventHandler>>(
-    regularArrayDOMEventHandlerNames,
-    identity,
-    identity,
-    ppCreate,
-  )
+type RawJavaScript = string
+type EventHandlerValues = { [eventHandlerName: string]: RawJavaScript }
+
+function useGetEventHandlerInfo(): EventHandlerValues {
+  const multiselectedProps = useGetMultiselectedProps(ppCreate, regularArrayDOMEventHandlerNames)
+  let result: EventHandlerValues = {}
 
-  const valueKeys = Object.keys(value)
+  forEachValue((values, eventHandlerName) => {
+    if (values.length === 1) {
+      const eitherValue = values[0]
+      if (isRight(eitherValue) && isJSXAttributeOtherJavaScript(eitherValue.value)) {
+        result[eventHandlerName] = eitherValue.value.javascript
+      }
+    }
+  }, multiselectedProps)
+
+  return useKeepReferenceEqualityIfPossible(result)
+}
+
+const EventHandlerSectionRow = betterReactMemo(
+  'EventHandlersSectionRow',
+  (props: { eventHandlerName: string; value: string }) => {
+    const { eventHandlerName, value } = props
+
+    const { onContextUnsetValue } = useInspectorContext()
+    const onUnsetValue = React.useCallback(
+      () => onContextUnsetValue([ppCreate(eventHandlerName)], false),
+      [eventHandlerName, onContextUnsetValue],
+    )
+
+    const eventHandlersContextMenuItems = utils.stripNulls([
```

Comment:
Looks like this should be wrapped in `useMemo`.

Reply:
Done